### ISSUE TITLE
Feed creation always has planned changes

### ISSUE THREAD

**Issue by ecca** [OctopusDeployLabs/terraform-provider-octopusdeploy#142]
**Describe the bug**
Provider seems to expect that `download_attempts` , `download_retry_backoff_seconds` and `is_enhanced_mode` is set for `feed_type = "Docker"`

**Steps to reproduce**
1. Setup provider
2. Create a feed
3. Run `terraform plan` again after created feed, planned changes should occur

```javascript
resource "octopusdeploy_feed" "docker_hub" {
  name                           = "DockerHub"
  feed_type                      = "Docker"
  feed_uri                       = "https://index.docker.io"
}
```

**Expected behavior**
When created, planned changes should not occur everytime on unchanged data.

**Logs and other supporting information**
```javascript
  # module.octopusdeploy.octopusdeploy_feed.docker_hub will be updated in-place
  ~ resource "octopusdeploy_feed" "docker_hub" {
      ~ download_attempts                     = 0 -> 5
      ~ download_retry_backoff_seconds        = 0 -> 10
        id                                    = "Feeds-1020"
      ~ is_enhanced_mode                      = false -> true
        name                                  = "DockerHub"
        # (5 unchanged attributes hidden)
    }
```

**Environment and versions:**
- OS: Windows
- Octopus Server Version: 2020.5 (Build 280)
- Terraform Version: v0.14.8
- Octopus Terraform Provider Version: v0.7.13

**Comment by jbristowe** [comment]
Thanks for submitting this issue!

I have noticed that this behaviour is the result of the way feeds are being serialised in our Go API client ([go-octopusdeploy](https://github.com/OctopusDeploy/go-octopusdeploy)), which is used by the provider. A fix will require adding models to the Go API client: [Enhancement: Disambiguate Feed Types #46](https://github.com/OctopusDeploy/go-octopusdeploy/issues/46). Once this is done, the provider will have to be updated with new resource types for each feed (i.e. `octopusdeploy_docker_container_registry`).

**Comment by ecca** [comment]
Thank you for looking into this. I've subscribed to the newly created enhancement with excitement 😄

**Comment by jbristowe** [comment]
This feature has been implemented [v0.7.15](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/releases/tag/v0.7.15), which is available through the [Terraform Registry](https://registry.terraform.io/providers/OctopusDeployLabs/octopusdeploy/latest).
